### ISSUE TITLE
Import / Export current configuration

### ISSUE THREAD

**Issue by agatim** [Noovolari/leapp#7]
It would be great if there was the possibility to export workspace.json (that contains all the personal configuration), and import it in another installation.

**Comment by andreacavagna01** [comment]
We will add this to our roadmap.
We have to ensure this feature with the feature of securing the credentials in the OS password manager.
Thanks for reaching us!

**Comment by Nuru** [comment]
@andreacavagna01 Any update on this? It is a blocker for us recommending it to clients because of the difficulty of distributing it to a large group. We need to be able to distribute it with pre-configured Sessions

**Comment by pethron** [comment]
We are trying to figure out a way to let people back up and restore the configuration file in case of a computer change. This is a bit difficult for the data in the keychain that we would like to avoid exporting due to possible leaking. Any help on this is really appreciated.

We will not implement import and export functionalities as requested by @Nuru because it's a use-case specifically designed for team and company-wide collaboration. We are working on a solution specifically designed for teams and enterprises that we will release in the next months. In the meantime, you can rely on AWS SSO integrations (and more to come).

**Comment by jcogilvie** [comment]
Adding my use case here as it is related:

I have a dozen chained-role profiles that all rely on one base user profile.  I would like to share the chained-role profiles, and let my teammates 'rebase' on their own user profile.  

This is simple to do with aws credentials files (since all they need to do is swap out their own keys on the base profile), but difficult to do in leapp.

**Comment by Nuru** [comment]
@andreacavagna01 Are you open to reconsidering the decision not to provide some kind of import support? 

Our use case is adding and removing account access for people already using Leapp and working on multiple accounts (freelancers, contractors, design shops, etc.).

What we would like is to be able to provide current Leapp users with a simple file (INI, YAML, or JSON) that describes a single session and optionally a single IdP connection so they can import it into their existing Leapp configuration. The fact that we can do it with the Leap CLI means that you are not going to force us to buy the pro version to accomplish this, but instead you are going to force us to install the CLI, which is a different kind of pain point.

**Comment by kevcube** [comment]
@andreacavagna01 no, the scriptable CLI is still not a replacement for being able to export all existing config in some file that can be swapped to a new machine or a team member.

Regarding @pethron's concern over keychain data, this is not necessary to include in the export, or could at least be optional.

I just need to export a bunch of chained AWS role configurations, but don't see a way to do this besides copying them 1-by-1. There are no relevant secrets, I would be fine grabbing a new SAML token on my next machine. But the non-secret stuff is a drag to copy over.

**Comment by Nuru** [comment]
This would also provide a way to work around #522

**Comment by Gowiem** [comment]
@andreacavagna01 I think this issue becomes more relevant now that Leapp Pro is going to be sunset. Is there anyway that we could get support for the "Import YAML" feature in Leapp Pro into to open source version? Now that my team and I are migrating off of Pro, we're realizing this is going to be problematic for us in onboarding new team members quickly. 

Wondering if we could open source the pro code for this feature and a community member could look into adding it?

**Comment by ericvilla** [comment]
Hi @Gowiem, with the last release the export Leapp Pro workspace functionality is now stable. It is documented [here](https://docs.leapp.cloud/latest/leapp-pro/export-pro-workspace/)

**Comment by Gowiem** [comment]
@ericvilla yeah, we went down that path and decided not to use that feature because it is only one workspace or one AWS org's set of roles and we are managing multiple orgs so it would overwrite our existing org roles. 

What we are really looking for is a declarative way to describe the roles that we want for an organization and then be able to import that into Leapp community edition easily. The YAML structure in Leapp Pro looks perfect for that, we just need the import functionality within the desktop app.
